src/ol/structs/RBush.js

```diff
@@ -233,7 +233,8 @@ class RBush {
   concat(rbush) {
     this.rbush_.load(rbush.rbush_.all());
     for (const i in rbush.items_) {
-      this.items_[i | 0] = rbush.items_[i | 0];
+      const x = Number(i);
+      this.items_[x | 0] = rbush.items_[x | 0];
```

Comment:
We shouldn't need the bitwise or here anymore (we aren't intentionally truncating to integer, and the value is already a number).

Reply:
If we remove those, `tsc` doesn't complain at all and no casts are necessary.